### PR TITLE
Add Steam Blast Furnace

### PR DESCRIPTION
**Pull request by Irgendwer01**
**What:**
This PR adds an Steam Blast Furnace which is basically an upgraded Primitive Blast Furnace that uses steam instead of coal/charcoal/coke and can process 4 Iron at the same time.

**Possible compatibility issue:**
None I can aware of.

### CODE REVIEW

**Comment by Syrcan** [comment]
I suppose this is meant as a replacement for the Industrial Primitive Blast Furnace (IPBF) in Gregicality, but are there other compelling arguments for this multiblock?
This looks completely overpowered, eliminating the need for coal/charcoal, while being much lower energy cost than the EBF, and not having to worry about oxygen either.
To solve the "steel issue", I would rather see the PBF be automatable, or the EBF recipe be simplified (as supplying 1 bucket of oxygen per steel is a pain early on).

**Comment by Irgendwer01** [comment]
> I suppose this is meant as a replacement for the Industrial Primitive Blast Furnace (IPBF) in Gregicality, but are there other compelling arguments for this multiblock? This looks completely overpowered, eliminating the need for coal/charcoal, while being much lower energy cost than the EBF, and not having to worry about oxygen either. To solve the "steel issue", I would rather see the PBF be automatable, or the EBF recipe be simplified (as supplying 1 bucket of oxygen per steel is a pain early on).

@Syrcan 
While the Steam Blast Furnace needs less Energy and can produce 4 ingots at the same time, it needs 56 seconds for each craft with normal iron and wrought iron needs 23 seconds, so I don't think it's that overpowered.

File: src/main/java/gregtech/common/metatileentities/MetaTileEntities.java [hunk 1]
```diff
@@ -530,6 +532,7 @@ public static void init() {
 
         STEAM_OVEN = registerMetaTileEntity(1023, new MetaTileEntitySteamOven(gregtechId("steam_oven")));
         STEAM_GRINDER = registerMetaTileEntity(1024, new MetaTileEntitySteamGrinder(gregtechId("steam_grinder")));
+        STEAM_BLAST_FURNACE = registerMetaTileEntity(1025, new MetaTileEntitySteamBlastFurnace(gregtechId("steam_blast_furnace")));
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/common/metatileentities/MetaTileEntities.java:535]: Conflicting ids

File: src/main/java/gregtech/common/metatileentities/multi/steam/MetaTileEntitySteamBlastFurnace.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+public class MetaTileEntitySteamBlastFurnace extends RecipeMapSteamMultiblockController {
+
+    private static final MultiblockAbility<?>[] ALLOWED_ABILITIES = {
+            MultiblockAbility.STEAM_IMPORT_ITEMS, MultiblockAbility.STEAM_EXPORT_ITEMS, MultiblockAbility.STEAM
+    };
+
+    public MetaTileEntitySteamBlastFurnace(ResourceLocation metaTileEntityId) {
+        super(metaTileEntityId, RecipeMaps.STEAM_BLAST_FURNACE_RECIPES, CONVERSION_RATE);
+        this.recipeMapWorkable = new SteamMultiWorkable(this, CONVERSION_RATE, 4);
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/common/metatileentities/multi/steam/MetaTileEntitySteamBlastFurnace.java:34]: This needs fixing after Trouser's Parallel PR was merged in, and also needs to set the parallel limit in the new way after Tech's PR.
`this.recipeMapWorkable.setParallelLimit(4)`

File: src/main/java/gregtech/common/metatileentities/multi/steam/MetaTileEntitySteamBlastFurnace.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Override
+    public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
+        return ConfigHolder.U.steelSteamMultiblocks ? SOLID_STEEL_CASING : BRONZE_PLATED_BRICKS;
+    }
+
+    @Nonnull
+    @Override
+    protected OrientedOverlayRenderer getFrontOverlay() {
+        return Textures.PRIMITIVE_BLAST_FURNACE_OVERLAY;
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/common/metatileentities/multi/steam/MetaTileEntitySteamBlastFurnace.java:66]: We should probably make a new FrontOverlay for this, incase texturepack makers want to modify things. You can simply reuse the current primitive blast furnace textures though, to have that texture there incase someone makes a new one.

File: src/main/resources/assets/gregtech/lang/en_us.lang [hunk 4]
```diff
@@ -9,6 +9,9 @@ death.attack.heat.player=%s was boiled alive by %s.
 
 enchantment.disjunction=Disjunction
 
+gregtech.machine.steam_blast_furnace.name=Steam Blast Furnace
+gregtech.machine.steam_blast_furnace.tooltip=Processes up to 4 Iron per craft. Requires §eSteam Hatches and Buses
+gregtech.multiblock.steam_blast_furnace.description=A upgrade to the Primitive Blast Furnace. Cannot use normal Input/Output busses, nor Fluid Hatches other than the Steam Hatch.
```
Comment (ALongStringOfNumbers) [inline src/main/resources/assets/gregtech/lang/en_us.lang:14]: `An upgrade...`

File: src/main/java/gregtech/integration/jei/multiblock/infos/SteamBlastFurnaceInfo.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                    .where('S', MetaTileEntities.STEAM_HATCH, EnumFacing.EAST)
+                    .build());
+        }
+        return Lists.newArrayList(shapeInfo);
+    }
+
+    @Override
+    public String[] getDescription() {
+        return new String[]{I18n.format("gregtech.multiblock.steam_blast_furnace.description")};
+    }
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/integration/jei/multiblock/infos/SteamBlastFurnaceInfo.java:59]: You could add some block tooltips here for the JEI preview for the steam hatches, clarifying that only steam hatches/buses can be used.

**Review by ALongStringOfNumbers — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/common/metatileentities/multi/steam/MetaTileEntitySteamBlastFurnace.java (hunk 2 above)
Comment (Irgendwer01) [reply src/main/java/gregtech/common/metatileentities/multi/steam/MetaTileEntitySteamBlastFurnace.java]: Fixed

File: src/main/java/gregtech/common/metatileentities/MetaTileEntities.java (hunk 1 above)
Comment (Irgendwer01) [reply src/main/java/gregtech/common/metatileentities/MetaTileEntities.java:535]: Fixed, but adds a small ID Shift that only affects large miners

File: src/main/resources/assets/gregtech/lang/en_us.lang (hunk 4 above)
Comment (Irgendwer01) [reply src/main/resources/assets/gregtech/lang/en_us.lang]: Fixed

**Pull request closed without merging**